marklogic-data-hub/src/main/java/com/marklogic/hub/deploy/util/ModuleWatchingConsumer.java

```diff
@@ -15,15 +18,36 @@
 public class ModuleWatchingConsumer extends LoggingObject implements Consumer<Set<Resource>> {
 
     private CommandContext commandContext;
+    private HubConfig hubConfig;
+    private Versions versions;
+
     private Command generateFunctionMetadataCommand;
 
     public ModuleWatchingConsumer(CommandContext commandContext, Command generateFunctionMetadataCommand) {
         this.commandContext = commandContext;
         this.generateFunctionMetadataCommand = generateFunctionMetadataCommand;
     }
 
+    /**
+     * This constructor is needed for when a user invokes "hubInit" when using the DH Gradle plugin, as the plugin
+     * won't be able to create a modules DatabaseClient yet as no username/password have been defined.
+     *
+     * @param commandContext
+     * @param hubConfig
+     * @param versions
+     */
+    public ModuleWatchingConsumer(CommandContext commandContext, HubConfig hubConfig, Versions versions) {
+        this.commandContext = commandContext;
+        this.hubConfig = hubConfig;
+        this.versions = versions;
+    }
+
     @Override
     public void accept(Set<Resource> resources) {
+        if (generateFunctionMetadataCommand == null && hubConfig != null && versions != null) {
+            generateFunctionMetadataCommand = new GenerateFunctionMetadataCommand(hubConfig.newModulesDbClient(), versions);
+        }
```

Comment:
1. This 'generateFunctionMetadataCommand' object is not managed by spring container whereas all other objects are. Will there be any issues because of that ?